oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,235 @@
+    export const MainNav = _MainNav;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+Now if we want to customize our _Header component, we can easily do it like
+
+.. code-block:: js
+
+    // SiteLogo being updated in Header
+    class MyThemedHeader extends _Header {
+        public render() {
+            return (
+                <header>
+                    {/* Replace <SiteLogo/> with a fancy widget */}
+                    <MyCustomAnimatedLogoWidget/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+    // Custom theme:
+    export const Header = MyThemedHeader;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+* We will provide support to add data or other components by using `props.children`. The `props.children` property is provided from React and can be extended to include specialized children like left and right for a Pane component. An example of this can be
+
+.. code-block:: js
+
+    const Header = (props) => {
+        return (
+            <h1>{props.title}</h1>
+            {props.children}
+        )
+    }
+
+    // use
+    <Header title={"Open Edx"} >
+        <h4>Open Source MOOC platform </h4>
+    </Header>
+
+Another example where specialized children are passed
+
+.. code-block:: js
+
+    const Pane = (props) => {
+        return (
+            <div className="pane">
+                <div className="pane-left">
+                    {props.left}
+                </div>
+                <div className="pane-right">
+                    {props.right}
+                </div>
+            </div>
+
+        )
+    }
+
+    // use
+    <Pane left={<Image />} right={<Explanation />} />
+
+
+* We will use function and placeholder to add additional content in customizable components. We can take an example of the above DefaultTheme and see _MainNav where it has support to add additional nav links by overriding `extraNavLinks` function.
+
+.. code-block:: js
+
+    // Customizable Main Navigation Area
+    class MyThemedNav extends _MainNav {
+        get extraNavLinks() {
+            return [
+                <a href="/about">About Us</a>,
+            ];
+        }
+    }
+
+    // Custom theme:
+    export const MainNav = MyThemedNav;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+* We will provide support for a global redux store, which will act as a central place to store data. This store will access to all data available to the system.
+
+* We will consider the layout of the data in the global store as an API. We will provide support to pre-fill the store with some common data like current user, current course, list of courses enrolled, etc. and add the less commonly needed data from REST API's using built-in redux actions. We will announce breaking changes if the layout of the data changes in store.
+
+
+* We will use containers to access data from the redux store and provide it to components via props.
```

Comment:
Can you elaborate on this? I don't think this is enough of an explanation. Containers are not a standard React concept, right?

Reply:
Yup, I don't think it's enough to say containers. Containers is a name people came up with for certain kinds of non-pure components that interact with external data like the redux store. It would be better to say that we will separate the redux data access from the component functionality, such that you can import the non-redux-connected component as well as a redux-connected version of the same component.